### DISCUSSION TITLE
Do I need to specifically add ORDER BY clause for aggregate function any or anyLast for default order?

### DISCUSSION THREAD

**Discussion by Alex-Cheng** [ClickHouse/ClickHouse#35492] [Q&A] [answered]
I want to use aggregate function `any` or `anyLast` in a query and depends on the default order (aka. the order specified by ORDER BY of table creation SQL of MergeTree engine). 

Do I need to specifically add ORDER BY clause for aggregate function `any` or `anyLast`? If I want to follow the default order and omit the ORDER BY clause like below SQL, will I ALWAYS get correct result?

```
create table t1(ID UInt64, name String, age UInt8, grp String) engine=MergeTree order by ID;
-- insert some data into the table
select any(ID) from t1 group by grp;   -- I want to follow the order of ID, like what defined in the CREATE SQL.
```

**Comment by Alex-Cheng** [comment]
Can I make sure the order of data by adding a subquery as shown as following?
```
select any(ID) from (select ID, grp from t1 order by ID) group by grp; 
```

**Reply by UnamedRus** [reply]
Yes, 

But i will suggest to look into argMax then, it will be cheaper.

**Reply by Alex-Cheng** [reply]
Thank you, master. I haven't notice that there is the very useful function, `argMax`.

**Reply by Alex-Cheng** [reply]
> But i will suggest to look into argMax then, it will be cheaper.

This is better answer.

**Comment by Alex-Cheng** [comment, marked as answer]
As suggested by @UnamedRus, argMax is the better way
